linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -115,6 +115,22 @@ void Process ()
 				ProcessQueue ();
 				ProcessVirtualMethods ();
 			}
+
+			// deal with [TypeForwardedTo] pseudo-attributes
+			foreach (AssemblyDefinition assembly in _context.GetAssemblies ()) {
+				if (!assembly.MainModule.HasExportedTypes)
+					continue;
+
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					if (!exported.IsForwarder)
+						continue;
+					var type = exported.Resolve ();
+					if (!Annotations.IsMarked (type))
+						continue;
+					Annotations.Mark (exported);
+					Annotations.Mark (assembly.MainModule);
```

Comment:
Found the commit where it was removed: https://github.com/xamarin/monotouch/commit/7c0e0a1df9276b52be92146ee2c7094167e4e2b4